packages/console/src/pages/TenantSettings/TenantBasicSettings/index.tsx

```diff
@@ -1,6 +1,120 @@
+import { type PatchTenant, type TenantInfo, TenantTag } from '@logto/schemas';
+import classNames from 'classnames';
+import { useCallback } from 'react';
+import { FormProvider, useForm } from 'react-hook-form';
+
+import AppError from '@/components/AppError';
+import AppLoading from '@/components/AppLoading';
+import PageMeta from '@/components/PageMeta';
+import SubmitFormChangesActionBar from '@/components/SubmitFormChangesActionBar';
+import UnsavedChangesAlertModal from '@/components/UnsavedChangesAlertModal';
+import useTenants from '@/hooks/use-tenants';
+
+import ProfileForm from './ProfileForm';
+import * as styles from './index.module.scss';
+import { type TenantSettingsForm } from './types.js';
+
 function TenantBasicSettings() {
-  // TODO @darcyYe implement this page
-  return <div>BasicSettings (WIP)</div>;
+  const {
+    api: cloudApi,
+    currentTenant: currentTenantInfo,
+    currentTenantId,
+    setTenants,
+    tenants,
+    error,
+    setError,
+    isLoading,
+  } = useTenants();
+
+  const methods = useForm<TenantSettingsForm>({
+    defaultValues: {
+      profile: {
+        name: currentTenantInfo?.name ?? 'My project',
+        tag: currentTenantInfo?.tag ?? TenantTag.Development,
+      },
+    },
+  });
+  const {
+    reset,
+    handleSubmit,
+    formState: { isDirty, isSubmitting },
+  } = methods;
+
+  const updateTenant = useCallback(
+    (tenant: TenantInfo) => {
+      const { id, name, tag } = tenant;
+      const index = tenants?.findIndex(({ id: tenantId }) => tenantId === id);
+      if (index !== undefined && index !== -1) {
+        setTenants([
+          ...(tenants ?? []).slice(0, index),
+          tenant,
+          ...(tenants ?? []).slice(index + 1),
+        ]);
```

Comment:
I think you can expose `mutate` from `useTenants` and you only need to call `mutate()` from here